resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -48,7 +47,7 @@
     private final InetSocketAddress nameServerAddr;
 
     private final boolean recursionDesired;
-    private volatile ScheduledFuture<?> timeoutFuture;
+    private volatile Future<?> timeoutFuture;
```

Comment:
@chrisvest I wonder if we want to back port some of the changes. Like use the more "narrow" type like done here.

Reply:
Made a PR: #11636